### PR TITLE
docs: update to match latest development

### LINKED ISSUE #485: Rework readme for annotation-based validator

### PR DESCRIPTION
**Pull request by lionel-nj**
closes #485 

**Summary:**

This PR updates `README.md` to match the latest development. 
**Expected behavior:** 

No code change

Please make sure these boxes are checked before submitting your pull request - thanks!

- ~[ ] Run the unit tests with `gradle test` to make sure you didn't break anything~
- ~[ ] Format the title like "Fix #<issue_number> - <short description of fix and changes>" (for example - "Fix #1111 - Check for null value before using field")~
- [x] Linked all relevant issues
- ~[ ] Include screenshot(s) showing how this pull request works and fixes the issue(s)~

### CODE REVIEW

**Comment by barbeau** [comment]
@lionel-nj I think the overall structure here looks good. I would add very brief details under the main README sections for "Build" (which points to downloading IntelliJ) and "Run" for what the basic command-line parameters are. The "How to run tests" could be moved out of the main README, and maybe into the contributing document.

File: BUILD.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Build Setup
+
+## Prerequisites for Gradle
+1. Clone this repository
+1. Install [Java Development Kit (JDK)](http://www.oracle.com/technetwork/java/javase/downloads/index.html)
```
Comment (fabrice-v) [inline BUILD.md:5]: v11+

File: CONTRIBUTION.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Contribution guidelines [WIP]
+
+## How to run tests locally
+This project includes unit and end-to-end tests in order to:
+1. Verify the implementation behaves as expected
```
Comment (fabrice-v) [inline CONTRIBUTION.md:5]: in tests as well as on real data

**Review by fabrice-v — COMMENTED** [review]
Shall we have a readme with links to all the new MD file collection?

Did we lose the CI badges? (first thing in README.md)

File: BUILD.md (hunk 1 above)
Comment (barbeau) [reply BUILD.md:5]: I believe the current annotation-based validator is actually JDK 8+.

**Comment by lionel-nj** [comment]
> Shall we have a readme with links to all the new MD file collection?
> 
> Did we lose the CI badges? (first thing in README.md)

We still have those: 
Capture d’écran, le 2020-12-03 à 18 30 29

**Review by barbeau — COMMENTED** [review]
@lionel-nj I started adding comments but because it included moving files and changes across multiple lines it was easier just to edit locally on my machine. I pushed my changes to this PR in https://github.com/MobilityData/gtfs-validator/pull/499/commits/db932a94cbb41e0612339ff0e39770ca2668321b.

Could you please take a look and see if you have any comments on these changes?

File: docs/ARCHITECTURE.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+- `SingleEntityValidator` for a given file are invoked and executed as soon as the file is loaded into memory (step 3)
+- `FileValidator` (for multiple files) are invoked and executed right after the whole GTFS archives is loaded into memory.
+ 
+5️⃣ **Notice export**
+1. Creates path to export notices as specified by command line input `--output` (or `-o`).
+1. Export notices in `NoticeContainer` as `.json` file. 
+
+🔚 **Output: validation result report** 
+
+### How to add a new validation rule _i.e._ a validator?
```
Comment (fabrice-v) [inline docs/ARCHITECTURE.md:61]: Does that go into architecture.md?

File: docs/CONTRIBUTIONS.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
+# Contribution guidelines [WIP]
```
Comment (fabrice-v) [inline docs/CONTRIBUTIONS.md:1]: Shouldn't this file be named CONTRIBUTING.md ?

File: docs/USAGE.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+Sample usage:
+
+``` 
+java -jar gtfs-validator-v2.0.jar --input relative/path/to/dataset --output relative/output/path --feed_name <name_of_the_feed> --threads <number_of_threads_to_use> 
+```
+
+...which will:
+ 1. Search for a GTFS dataset located at `relative/path/to/dataset`
+ 1. Validate the GTFS data and output the results to the directory located at `relative/output/path`. 
+ 1. Export the validation report to `JSON` by default. This folder will contain a single `.json` file with information related to the validation process. The validation report will be named as `report.json`.
```
Comment (fabrice-v) [inline docs/USAGE.md:16]: is that up to date?

Comment (lionel-nj) [reply docs/USAGE.md:16]: Yes it is :)

File: docs/ARCHITECTURE.md (hunk 3 above)
Comment (lionel-nj) [reply docs/ARCHITECTURE.md:61]: Goog point! I'll move this one out of the `ARCHITECTURE.md` and create `ADD_RULES.md`.

File: docs/USAGE.md [hunk 6]
```diff
@@ -0,0 +1,16 @@
+# Instructions to run the application locally
+1. Install [Java 8 or higher](https://www.oracle.com/java/technologies/javase-downloads.html)
```
Comment (fabrice-v) [inline docs/USAGE.md:2]: Mmmm, we agreed with Dmitri the supported version is LTS 11.2 @aababilov ?

**Review by fabrice-v — COMMENTED** [review]
I think minimum version of Java should be 11 and not 8?

File: docs/USAGE.md (hunk 6 above)
Comment (barbeau) [reply docs/USAGE.md:2]: There shouldn't be a difference given the current code in the project. The current code doesn't use any features introduced in JDK9+, so you can run it on JDK8. If we introduce code that requires greater than JDK8 then we could bump this. But to my knowledge there isn't a drawback of allowing this code to run on JDK8.

**Review by fabrice-v — APPROVED** [review]

**Pull request merged**